### PR TITLE
Jb47430

### PR DESCRIPTION
**Pull request by etaishev**
Move OS brand name to variable, which configure by qmake variable. By default use Sailfish OS, if variable VENDOR_NAME is set use $${VENDOR_NAME} OS

### CODE REVIEW

**Comment by martyone** [comment]
Sadly this touches code that has been changed a lot under branch jb45560 which is about to be merged soon. You will need to rebase you changes on top of that :(

**Comment by etaishev** [comment]
Okay, wait it.

**Comment by vigejolla** [comment]
I think we should have a better name for the qmake variable than "VENDOR_NAME". As the contents of the variable is not the name of the vendor, but the base name of the OS.

**Comment by etaishev** [comment]
Update on current master

**Comment by martyone** [comment]
@etaishev I only returned now to the review and I found some issues that should be fixed, so I reverted it from master. Please open a new PR with the same content, I will add my comments there.

**Comment by etaishev** [comment]
@martyone   https://github.com/sailfishos/sailfish-qtcreator/pull/226

File: src/plugins/mer/merconstants.h [hunk 1]
```diff
@@ -29,11 +29,19 @@
...
 namespace Constants {
 
+#ifdef MER_BASE_OS_NAME
+const char MER_OS_NAME[] = QT_TRANSLATE_NOOP("Mer", MER_BASE_OS_NAME " OS");
+const char MER_SDK_NAME[] = QT_TRANSLATE_NOOP("Mer", MER_BASE_OS_NAME " SDK");
+#else
+const char MER_OS_NAME[] = "Sailfish OS";
+const char MER_SDK_NAME[] = "Sailfish SDK";
+#endif
+
```
Comment (martyone) [inline src/plugins/mer/merconstants.h]: This seems wrong - one branch is i18n-enabled, the other is not. Is is necessary to enable i18n for these at all?

This also makes me think, do we really need to add a custom mechanism to change the branding? Wouldn't it be more convenient to use i18n for that. I mean you can leave Sailfish OS/Sailfish SDK in the code and provide translations (easy to automate that) that say something else.

Comment (martyone) [reply src/plugins/mer/merconstants.h]: Of course this comment should be posted under the new PR...

**Pull request merged**